### PR TITLE
doc(iot-dev): Add clarification to startDeviceTwin() API docs

### PR DESCRIPTION
**Pull request by timtay-microsoft**
See #772 

Customer's need a bit more help understanding how startDeviceTwin works, so I've beefed up the javadocs for it

### CODE REVIEW

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java [hunk 1]
```diff
@@ -463,7 +463,16 @@ public void getDeviceTwin() throws IOException
     }
 
     /**
-     * Starts the device twin.
+     * Starts the device twin. This device client will receive a callback with the current state of the full twin, including
+     * reported properties and desired properties. After that callback, this device client will receive a callback
+     * each time a desired property is updated. That callback will either contain the full desired properties set, or
+     * only the updated desired property depending on how the desired property was changed. IoT Hub supports a PUT and a PATCH
+     * on the twin, and the PATCH will cause this device client to receive the full desired properties set. Similarly, the version
```
Comment (abhipsaMisra) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java:470]: Shouldn't PATCH return only the patched update?

Comment (abhipsaMisra) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java:470]: From the github issue:

```
Initial desired state:
{ “a” : 1, “b” : 2 }

If customer sends PATCH { “a” : -1} device will receive { “a” : -1 }. If customer sends PATCH { “a”: 1, “b”: 2 } device will receive { “a”: 1, “b”: 2 }

Having said that, if customer sends PUT { “a” : 1, “c” : 100 }, device will get { “a”: 1, “b”: null, “c” : 100 }
```

Comment (timtay-microsoft) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java:470]: Interesting, that is what the service team said. I double checked to see how our SDK does desired property updates, and it is a PATCH that causes the device side to get the full desired property update. However, it looks like our service SDK sends the full desired properties set even when patching just one property. That's why our SDK's PATCH behaved like the PUT.

I'll update the javadoc here and the issue accordingly

**Comment by abhipsaMisra** [comment]
You should add your finding to the documentation for property update as well (that the API we call is a PATCH, but the SDK will send the entire twin object, instead of sending only the patch object)

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java [hunk 2]
```diff
@@ -463,7 +463,19 @@ public void getDeviceTwin() throws IOException
     }
 
     /**
-     * Starts the device twin.
+     * Starts the device twin. This device client will receive a callback with the current state of the full twin, including
+     * reported properties and desired properties. After that callback, this device client will receive a callback
```
Comment (abhipsaMisra) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java:467]: nit: After that callback _is received_ ...

**Pull request merged**